### PR TITLE
AI emote statusdisplays, radials and fixes

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

Kills the 13-10yr old code that is AI emotes. The Emote statusdisplays now work either through clicking them or using a *emote in the chatbar, much like slimes.

Also ports a sanity pr ensuring power disabled AIs can turn back on their apc

Ports:
- https://github.com/tgstation/tgstation/pull/55977
- https://github.com/tgstation/tgstation/pull/56255

## Why It's Good For The Game

Removes a 17 line long row of if statements. Turns another obfuscated verb into an interactive system.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/df9bdd2c-9665-4fdb-b7dc-8254acb192bc

</details>

## Changelog
:cl: rkz, coiax, zxaber
add: AI's can now set their emote displays across the station through a radial by clicking on them, or by using *emotes in chatbar.
refactor: Refactored the AI Emote system to not be a 17 line if statement column.
fix: Fix issue where AI couldnt change statusdisplay emote after it was set
fix: Added extra checks to ensure a power disabled AI could eventually return itself to power (reinforcing intended behavior)
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -75,7 +75,7 @@
 	var/turf/waypoint //Holds the turf of the currently selected waypoint.
 	var/waypoint_mode = FALSE		//Waypoint mode is for selecting a turf via clicking.
 	var/call_bot_cooldown = 0		//time of next call bot command
-	var/apc_override = FALSE		//hack for letting the AI use its APC even when visionless
+	var/obj/machinery/power/apc/apc_override //Ref of the AI's APC, used when the AI has no power in order to access their APC.
```
Comment (PowerfulBacon) [inline code/modules/mob/living/silicon/ai/ai.dm:78]: Is this reference properly cleared if the APC gets deleted?

Comment (Tsar-Salat) [reply code/modules/mob/living/silicon/ai/ai.dm:78]: Yes.
`/mob/living/silicon/ai/proc/start_RestorePowerRoutine()`, `/mob/living/silicon/ai/proc/ai_restore_power()` and `/obj/machinery/power/apc/ui_state(mob/user)` all have handling for if the AI's local area APC is not detected.

Comment (PowerfulBacon) [reply code/modules/mob/living/silicon/ai/ai.dm:78]: Oh, I see this was never a boolean to begin with, you just typed it properly

File: code/modules/mob/living/silicon/ai/login.dm [hunk 2]
```diff
@@ -1,11 +1,8 @@
 /mob/living/silicon/ai/Login()
 	..()
 	if(stat != DEAD)
-		for(var/each in GLOB.ai_status_displays) //change status
-			var/obj/machinery/status_display/ai/O = each
-			O.mode = 1
-			O.emotion = "Neutral"
-			O.update()
+		if(lacks_power() && apc_override) //Placing this in Login() in case the AI doesn't have this link for whatever reason.
+			to_chat(usr, "["<span class='warning'>Main power is unavailable, backup power in use. Diagnostics scan complete.</span>"] <A HREF='?src=[REF(src)];emergencyAPC=[TRUE]'>Local APC ready for connection.</A>")
```
Comment (PowerfulBacon) [inline code/modules/mob/living/silicon/ai/login.dm]: This seems really weird to me, why is this substring inside a string interpolation?

Comment (Tsar-Salat) [reply code/modules/mob/living/silicon/ai/login.dm]: Its just a ref link the AI can click to access their local area apc. 

In our current code, if an AI loses power and their APC turns off via EMP or a rival silicon hitting the off switch, they will go through a repowering process before eventually being able to access and turn on their apc again. Given they are blind during the period they cannot access it themselves and must run through the process.

However, this can be interrupted by a logout/client loss and the thing they are supposed to be interacting with may not appear.

Basically, we are waiting for two circumstances to arise here for the AI. They have no power currently being supplied (making them unable to see or take any action) and have gone through the process of repowering for x minutes to be able to connect to their local apc to repower themselves. If both are true, upon relog, the AI player gets the ref in chat.

Effectively just a sanity measure.

Comment (PowerfulBacon) [reply code/modules/mob/living/silicon/ai/login.dm]: No, I mean "["string constant"]" should just be "string constant"

Comment (Tsar-Salat) [reply code/modules/mob/living/silicon/ai/login.dm]: Oh. Alright.

Comment (Tsar-Salat) [inline code/modules/mob/living/silicon/ai/login.dm]:
```suggestion
			to_chat(usr, ""<span class='warning'>Main power is unavailable, backup power in use. Diagnostics scan complete.</span>" <A HREF='?src=[REF(src)];emergencyAPC=[TRUE]'>Local APC ready for connection.</A>")
```

Comment (PowerfulBacon) [reply code/modules/mob/living/silicon/ai/login.dm]:
```suggestion
			to_chat(usr, "<span class='warning'>Main power is unavailable, backup power in use. Diagnostics scan complete.</span> <A HREF='?src=[REF(src)];emergencyAPC=[TRUE]'>Local APC ready for connection.</A>")
```

**Review by PowerfulBacon — APPROVED** [review]
